core/ev-admin.php

```diff
@@ -204,52 +215,66 @@ function update_videos_handler() {
     $delete = isset( $options['delete'] ) ? $options['delete'] : null;
 
     $new_messages = $trash_messages = '';
 
     // figure out whether we're updating a single author, or all
-    // and populate the $update_authors array
+    // if single limit the $update_authors and $update_hosts array accordingly
     if( isset( $_POST['author_id'] ) && isset( $_POST['host_id'] ) ) {
       // it's single
+      $this_author = $_POST['author_id'];
+      $this_host = $_POST['host_id'];
       $eligibles = array();
 
-      // get the relevant local author. not so simple in multidimensional!
-      foreach( $AUTHORS as $authorkey => $authordata ){
-        if( in_array( $_POST['host_id'], $authordata ) ) {
-          $eligibles[] = $authorkey;
+      // get the relevant local author. not so simple in multidimensional array
+      // problem is, authors are not indexed by host. look for host_id in values
+      // could be multiples
+      // *** TO DO MAYBE - break up $AUTHORS table to 'authors' under each $HOSTS ***
+      // Get all authors of a given host
+      foreach( $AUTHORS as $authorindex=>$authorarray ){
+        if( in_array( $this_host, $authorarray ) ) {
+          $eligibles[] = $authorindex;
         }
       }
 
       if( $eligibles ) {
         $thisindex = array();
-
-        foreach( $eligibles as $eligible => $originalkey ){
-          if( in_array( $_POST['author_id'], $AUTHORS[$originalkey] ) ) {
-            $thisindex[] = $originalkey;
+        // Check if $_POST['author_id'] matches author_id
+        foreach( $eligibles as $authorindex ){
+          if( in_array( $this_author, $AUTHORS[$authorindex] ) ) {
+            $thisindex[] = $authorindex;
           }
         }
-        $thisindex = array_shift( $thisindex );
-        $update_authors = array( $AUTHORS[$thisindex] );
+        $thisindex = array_shift( $thisindex ); // first one is the one
+        $update_authors = array( $AUTHORS[$thisindex] ); // has to stay loopable
+        $update_hosts = array( $this_host=>$HOSTS[$this_host] ); // has to stay indexed and loopable
         $single = true;
       }
 
     } else { // it's update all
       $update_authors = $AUTHORS;
+      $update_hosts = $HOSTS;
       $single = null;
     }
 
     // post_new_videos() gets everything new and returns messages about it
-    $new_messages = $this->post_new_videos( $update_authors, $HOSTS, $single );
+    $post_results = $this->post_new_videos( $update_authors, $update_hosts, $single );
+    $new_messages = $post_results['messages'];
+    $new_video_ids = $post_results['new_video_ids'];
 
     // trash_deleted_videos() checks for videos deleted on host and returns messages about it
-    if( $delete && !$single ) $trash_messages = $this->trash_deleted_videos( $update_authors, $HOSTS );
+    if( $delete ) {
+      $trash_messages = $this->trash_deleted_videos( $update_hosts, $new_video_ids );
+    }
 
     $messages = $new_messages . $trash_messages;
+    // */
```

Comment:
Yeah, I knew they were related - just looked like some left over debugging code. Not to worry! :-)